hyperparameter_tuning/pytorch_mnist/hpo_pytorch_mnist.ipynb

```diff
@@ -0,0 +1,369 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hyperparameter_ranges = {'lr': ContinuousParameter(0.001, 0.1),'batch-size': CategoricalParameter([32,64,128,256,512])}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next we'll specify the objective metric that we'd like to tune and its definition, which includes the regular expression (Regex) needed to extract that metric from the CloudWatch logs of the training job. In this particular case, our script emits average loss value and we will use it as the objective metric, we also set the objective_type to be 'minimize', so that hyperparameter tuning seeks to minize the objective metric when searching for the best hyperparameter setting. By default, objective_type is set to 'maximize'."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "objective_metric_name = 'average test loss'\n",
+    "objective_type = 'Minimize'\n",
+    "metric_definitions = [{'Name': 'average test loss',\n",
+    "                       'Regex': 'Test set: Average loss: ([0-9\\\\.]+)'}]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we'll create a `HyperparameterTuner` object, to which we pass:\n",
+    "- The PyTorch estimator we created above\n",
+    "- Our hyperparameter ranges\n",
+    "- Objective metric name and definition\n",
+    "- Tuning resource configurations such as Number of training jobs to run in total and how many training jobs can be run in parallel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tuner = HyperparameterTuner(estimator,\n",
+    "                            objective_metric_name,\n",
+    "                            hyperparameter_ranges,\n",
+    "                            metric_definitions,\n",
+    "                            max_jobs=9,\n",
+    "                            max_parallel_jobs=3,\n",
+    "                            objective_type=objective_type)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Launch hyperparameter tuning job\n",
+    "And finally, we can start our hyperprameter tuning job by calling `.fit()` and passing in the S3 path to our train and test dataset.\n",
+    "\n",
+    "After the hyperprameter tuning job is created, you should be able to describe the tuning job to see its progress in the next step, and you can go to SageMaker console->Jobs to check out the progress of the progress of the hyperparameter tuning job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tuner.fit({'training': inputs})"
+   ]
+  },
+  {
```

Comment:
Do you want to throw a `tuner.wait()` in here?  If users don't fully understand that tuner.fit() orchestrates everything in the background, and that they'll need to monitor the tuning job's progress separately, then they'll get an error when trying to run the next cell.  Including `.wait()` will prevent this.  It does mean their notebook kernel will be locked for ~20 minutes though, which might not be ideal.  I'll leave it up to you.  Thanks.